app/scripts/lib/channels/notifier.js

```diff
@@ -11,20 +11,56 @@ define(function (require, exports, module) {
 
   var _ = require('underscore');
   var Backbone = require('backbone');
+  var Validate = require('lib/validate');
 
   // Events that have the 'internal:' namespace should only be
   // handled by the content server. Other events may be handled
   // both externally and internally to the content server.
   var EVENTS = {
-    COMPLETE_RESET_PASSWORD_TAB_OPEN: 'fxaccounts:complete_reset_password_tab_open',
-    DELETE: 'fxaccounts:delete',
-    PROFILE_CHANGE: 'profile:change',
-    SIGNED_IN: 'internal:signed_in',
-    SIGNED_OUT: 'fxaccounts:logout'
+    COMPLETE_RESET_PASSWORD_TAB_OPEN: {
+      name: 'fxaccounts:complete_reset_password_tab_open',
+      schema: null
+    },
+    DELETE: {
+      name: 'fxaccounts:delete',
+      schema: { uid: 'String' }
+    },
+    PROFILE_CHANGE: {
+      name: 'profile:change',
+      schema: { uid: 'String' }
+    },
+    SIGNED_IN: {
+      name: 'internal:signed_in',
```

Comment:
> I may be splitting hairs here, but I disagree. The term command there seems synonymous with the term event from this module, not name.

I didn't realize before this we use two different names in different locations for the same concept... That's a problem.

`command` came about because that's what the browser uses for both the [original FirefoxAccountsCommand message](https://dxr.mozilla.org/mozilla-central/rev/8cb42e7a16b42162c9930f37b9e1f820c2eb126b/browser/base/content/aboutaccounts/aboutaccounts.js#277) and now the [WebChannel message](https://dxr.mozilla.org/mozilla-central/rev/8cb42e7a16b42162c9930f37b9e1f820c2eb126b/services/fxaccounts/FxAccountsWebChannel.jsm#145). `event.name` mirrors the DOM approach so there is precedent there too.

You are correct that `command` is only semantically correct some of the time. Some messages are commands, others are requests and still others are simply informative. Still, I find grep-driven development easiest if the same ill-fitting name is used everywhere.